### ISSUE TITLE
App is artificially locked to portrait mode although this is an accessibility issue

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-app-android#5355]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Huawei P Smart 2019
- Android version: 10
- App version: 2.24.2

## Describe the bug

The CWA Android app is artificially locked to only display the UI in portrait mode, although the app supports landscape mode, as you can easily confirm by using two apps side by side in landscape mode. Here is a screenshot: 

## Steps to reproduce the issue

1. Open CWA
2. Rotate the device to landscape orientation
3. Notice that the app does not respond to the rotation change

## Expected behaviour

The UI of the app needs to be available in landscape mode, as only having the UI in portrait mode is an accessibility issue, as you yourself identified in the [Accessibility Statement by the Robert Koch Institute and the Federal Ministry of Health](https://www.coronawarn.app/en/accessibility/#android), where it says:

> The content of the app is limited to portrait display, in contradiction to requirement 1.3.4 WCAG 2.1 or 11.1.3.4 EN 301 549 V3.2.1.

## Possible Fix

Remove the lock which hinders the app to render the UI in landscape mode.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

- The enhancement request https://github.com/corona-warn-app/cwa-app-android/issues/4309 also calls for support of landscape mode. Unfortunately this request was not mirrored to jira, so there was never any feedback on it.

- Allowing landscape mode will exposure some formatting issues which are not noticeable in portrait mode e.g. the cut off head in the following:

![image](https://user-images.githubusercontent.com/66998419/158818952-d8520439-5b0a-467d-944f-f536bbe989b3.png)

- Other functions, like the display of a QR code for scanning do not work in landscape mode because the app does not display the complete QR code in landscape orientation. It is either cut off at the top or the bottom.

- These formatting and layout issues would need to be identified and resolved before allowing the app to rotate to landscape mode.

If you want to try this out yourself refer to https://developer.android.com/guide/topics/manifest/activity-element in the section `android:screenOrientation`. 

The settings to allow or disallow landscape mode are in the [AndroidManifest.xml](https://github.com/corona-warn-app/cwa-app-android/blob/main/Corona-Warn-App/src/main/AndroidManifest.xml) file.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

> Allowing landscape mode will exposure some formatting issues which are not noticeable in portrait mode e.g. the cut off head in the following:

That is true, however, in my opinion some cut off illustrations, etc. should
a) not stop the team from updating the app to support landscape orientation and ship bug fixes for this in later updates
b) be easily fixable

> Other functions, like the display of a QR code for scanning do not work in landscape mode because the app does not display the complete QR code in landscape orientation. It is either cut off at the top or the bottom.

As you can tap on the QR code & then a full screen view of the QR code is opened, this is not a problem neither. Yes, the QR code should be displayed completely instantly, but with the tap on the QR code workaround it is for an initial release with landscape orientation support OK.

**Comment by MikeMcC399** [comment]
@Ein-Tim 

I would view support of landscape mode as a major enhancement for the app. It's very easy to allow the app to rotate, however to allow it to display information in a useful way in landscape mode would require development of new layouts. I wouldn't see that as a simple bug fixing exercise. It that were the case, then probably the developers would have enabled landscape mode a long time ago. Perhaps anyway they will be able to provide some feedback.

I didn't make any attempt to go through all functions of the app when I mentioned a couple of examples where the layout has issues in landscape mode.

https://developer.android.com/guide/topics/large-screens/support-different-screen-sizes gives some insight into the problems involved.

**Comment by Ein-Tim** [comment]
@MikeMcC399 I think there are two ways of integrating landscape mode. Either, the team just allows the app to rotate and fixes the issue where illustrations or text is cut off (this would be the easy way), but for sure the app could be optimized so that the form factor landscape is used in the design and new layouts are created (more workload).

Thanks for giving your feedback on this issue! I was just surprised that the app actually works in landscape if it is used in split scree, so I opened this issue.

**Comment by fynngodau** [comment]
CCTG has [lifted the orientation lock to enable landscape mode](https://codeberg.org/corona-contact-tracing-germany/cwa-android/issues/86).

* The issue that is seen in https://github.com/corona-warn-app/cwa-app-android/issues/5355#issuecomment-1186263352 can be fixed easily by replacing `android:scaleType="centerCrop"` with `"centerInside"` as in https://codeberg.org/corona-contact-tracing-germany/cwa-android/commit/31caed07c681d77e33060ccefdc841645a45b584.
* [There is one additional issue with the entry of TANs in landscape](https://codeberg.org/corona-contact-tracing-germany/cwa-android/issues/94). [We fixed this to some extent](https://codeberg.org/corona-contact-tracing-germany/cwa-android/commit/dc71097d25b9a74088d429260f0bc01ceeb3b0ab), but UX is not optimal as it may scroll the screen such that the fields are invisible.
* We have no solution for the mentioned QR code issue, which can in theory also appear on special portrait devices with fun aspect ratios: https://codeberg.org/corona-contact-tracing-germany/cwa-android/issues/160 and https://codeberg.org/corona-contact-tracing-germany/cwa-android/issues/219.

Besides that the layout is obviously not optimized for landscape, no other problems have been found so far.

**Comment by larswmh** [comment]
Thanks for your report @Ein-Tim. We have created an internal ticket for it and will raise this topic internally.
Internal Tracking ID: [EXPOSUREAPP-13641](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13641)

---
Corona-Warn-App Open Source Team
